### PR TITLE
feat: add workspace auto destroy duration field

### PR DESCRIPTION
**Pull request by notchairmk**
## Description

Adds an `auto_destroy_activity_duration` field to Workspaces (see [API docs](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/workspaces#create-a-workspace:~:text=auto-destroy-activity-duration)). Makes use of `jsonapi.NullableAttr` because this is an optional string field, but we will want to be able to send an explicit null to unset.

## Testing plan

1. Run auto destroy tests, confirm that they're valid `envchain go-tfe go test -run "TestWorkspacesAutoDestroy.*"  -v ./...`

## Output from tests
Including output from tests may require access to a TFE instance. Ignore this section if you have no environment to test against.

```
$ TFE_ADDRESS="https://example" TFE_TOKEN="example" go test ./... -v -run "TestWorkspacesAutoDestroy.*"

?   	github.com/hashicorp/go-tfe/examples/backing_data	[no test files]
?   	github.com/hashicorp/go-tfe/examples/configuration_versions	[no test files]
?   	github.com/hashicorp/go-tfe/examples/organizations	[no test files]
?   	github.com/hashicorp/go-tfe/examples/registry_modules	[no test files]
?   	github.com/hashicorp/go-tfe/examples/run_errors	[no test files]
?   	github.com/hashicorp/go-tfe/examples/state_versions	[no test files]
?   	github.com/hashicorp/go-tfe/examples/users	[no test files]
?   	github.com/hashicorp/go-tfe/examples/workspaces	[no test files]
?   	github.com/hashicorp/go-tfe/mocks	[no test files]
=== RUN   TestWorkspacesAutoDestroy
--- PASS: TestWorkspacesAutoDestroy (4.72s)
=== RUN   TestWorkspacesAutoDestroyDuration
--- PASS: TestWorkspacesAutoDestroyDuration (3.85s)
PASS
ok  	github.com/hashicorp/go-tfe	9.099s
...
```

### CODE REVIEW

File: workspace_integration_test.go [hunk 1]
```diff
@@ -2747,7 +2748,7 @@ func TestWorkspacesAutoDestroy(t *testing.T) {
 	})
 	t.Cleanup(wCleanup)
 
-	require.Equal(t, wTest.AutoDestroyAt, autoDestroyAt)
+	require.Equal(t, autoDestroyAt, wTest.AutoDestroyAt)
```
Comment (notchairmk) [inline workspace_integration_test.go:2751]: swapped the order on these since require wants `require.Equal(t, <expected>, <actual>)`

File: workspace.go [hunk 2]
```diff
@@ -513,6 +517,9 @@ type WorkspaceUpdateOptions struct {
 	// Optional: The time after which an automatic destroy run will be queued
 	AutoDestroyAt jsonapi.NullableAttr[time.Time] `jsonapi:"attr,auto-destroy-at,iso8601,omitempty"`
 
+	// Optional: The period of time to wait after workspace activity to trigger a destroy run.
```
Comment (brandonc) [inline workspace.go:520]: Can we describe the format of this field? Is it a go-style duration string using the time unit suffix?

**Comment by brandonc** [comment]
Is it the intention that a tfe_workspace without a auto_destroy_at config should unset any drifted autodestroy setting?

Usually this kind of drift correction is good but in the case of auto_destroy, it seems like terraform wouldn't be the best source of that config.

Methodology:
1. create workspace using basic config with no auto destroy
2. set up auto destroy in the UI
3. terraform apply

Outcome:
terraform plans to unset the auto destroy setting

Screenshot 2024-05-21 at 2 14 26 PM

**Comment by notchairmk** [comment]
> Is it the intention that a tfe_workspace without a auto_destroy_at config should unset any drifted autodestroy setting?
> 
> Usually this kind of drift correction is good but in the case of auto_destroy, it seems like terraform wouldn't be the best source of that config.
> 
> Methodology:
> 
> 1. create workspace using basic config with no auto destroy
> 2. set up auto destroy in the UI
> 3. terraform apply
> 
> Outcome: terraform plans to unset the auto destroy setting
> 
> Screenshot 2024-05-21 at 2 14 26 PM

I think this is maybe not the right PR for this comment. Isn't this exactly the same behavior if you're setting another attribute on the workspace like `description`?

**Comment by brandonc** [comment]
I totally put this in the wrong PR. Let me try again.

**Review by brandonc — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,6 @@
 # Unreleased
+## Enhancements
+* Adds the `AutoDestroyActivityDuration` field to `Workspace` by @notchairmk [#900](https://github.com/hashicorp/go-tfe/pull/900)
```
Comment (emailnitram) [inline CHANGELOG.md:3]: shouldn't this be 902?
